### ISSUE TITLE
grpcproxy: TestV3LeaseRequireLeader: did not receive leader loss error

### ISSUE THREAD

**Issue by heyitsanthony** [coreos/etcd#7952]
vi jenkins: https://jenkins-etcd-public.prod.coreos.systems/job/etcd-proxy/1637/console
```
--- FAIL: TestV3LeaseRequireLeader (0.98s)
	v3_lease_test.go:532: did not receive leader loss error
	v3_lease_test.go:527: err = context canceled, want etcdserver: no leader
```

**Comment by gyuho** [comment]
Also from https://semaphoreci.com/coreos/etcd/branches/pull-request-7949/builds/2

```
--- FAIL: TestV3LeaseRequireLeader (3.04s)
	v3_lease_test.go:532: did not receive leader loss error
	v3_lease_test.go:527: err = rpc error: code = Canceled desc = context canceled, want etcdserver: no leader
```

**Comment by gyuho** [comment]
@heyitsanthony 

This is fairly easy to reproduce without grpc-proxy.

I added second `select` statement to see if it receives `ErrNoLeader` if it waited more, and it did. The server was just being slow, so it returned `ErrNoLeader` in +800ms, while the timeout is 500ms.

```
[DEBUG] waiting 500ms
2017-05-18 22:21:26.742832 I | etcdserver/api/v3rpc: transport: http2Client.notifyError got notified that the client transport was broken EOF.
2017-05-18 22:21:26.743122 I | etcdserver/api/v3rpc: grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix localhost:57586252471356315960: connect: no such file or directory"; Reconnecting to {localhost:57586252471356315960 <nil>}
2017-05-18 22:21:26.743575 W | rafthttp: lost the TCP streaming connection with peer ff14b166cbcbf7e (stream Message reader)
2017-05-18 22:21:26.743661 W | rafthttp: lost the TCP streaming connection with peer ff14b166cbcbf7e (stream MsgApp v2 reader)
2017-05-18 22:21:26.750860 E | rafthttp: failed to write ff14b166cbcbf7e on stream Message (write unix 127.0.0.1:2100113443->@: write: broken pipe)
2017-05-18 22:21:26.750995 I | rafthttp: peer ff14b166cbcbf7e became inactive
2017-05-18 22:21:26.752592 W | rafthttp: lost the TCP streaming connection with peer ff14b166cbcbf7e (stream Message writer)
2017-05-18 22:21:26.752795 E | rafthttp: failed to write a59d378281242d56 on stream Message (write unix 127.0.0.1:2100113443->@: write: broken pipe)
2017-05-18 22:21:26.752875 I | rafthttp: peer a59d378281242d56 became inactive
2017-05-18 22:21:26.754877 W | rafthttp: lost the TCP streaming connection with peer a59d378281242d56 (stream Message writer)
2017-05-18 22:21:26.812630 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 31.87671ms)
2017-05-18 22:21:26.812737 W | etcdserver: server is likely overloaded
2017-05-18 22:21:26.812804 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 32.071387ms)
2017-05-18 22:21:26.812862 W | etcdserver: server is likely overloaded
2017-05-18 22:21:26.887037 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 54.21745ms)
2017-05-18 22:21:26.887133 W | etcdserver: server is likely overloaded
2017-05-18 22:21:26.887199 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 54.587377ms)
2017-05-18 22:21:26.887254 W | etcdserver: server is likely overloaded
2017-05-18 22:21:26.962740 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 42.008736ms)
2017-05-18 22:21:26.962837 W | etcdserver: server is likely overloaded
2017-05-18 22:21:26.962909 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 42.201391ms)
2017-05-18 22:21:26.962960 W | etcdserver: server is likely overloaded
2017-05-18 22:21:27.061162 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 78.401153ms)
2017-05-18 22:21:27.061335 W | etcdserver: server is likely overloaded
2017-05-18 22:21:27.061463 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 78.547114ms)
2017-05-18 22:21:27.061555 W | etcdserver: server is likely overloaded
2017-05-18 22:21:27.133134 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 32.046693ms)
2017-05-18 22:21:27.133225 W | etcdserver: server is likely overloaded
2017-05-18 22:21:27.133286 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 32.237352ms)
2017-05-18 22:21:27.133337 W | etcdserver: server is likely overloaded
2017-05-18 22:21:27.209064 W | raft: 18726fe191b3281c stepped down to follower since quorum is not active
2017-05-18 22:21:27.209212 I | raft: 18726fe191b3281c became follower at term 2
2017-05-18 22:21:27.209282 I | raft: raft.node: 18726fe191b3281c lost leader 18726fe191b3281c at term 2
[DEBUG] FAIL!!!!!!!!!! took 500.221983ms (waiting more)
2017-05-18 22:21:27.360679 I | raft: 18726fe191b3281c is starting a new election at term 2
2017-05-18 22:21:27.360847 I | raft: 18726fe191b3281c became candidate at term 3
2017-05-18 22:21:27.360918 I | raft: 18726fe191b3281c received MsgVoteResp from 18726fe191b3281c at term 3
2017-05-18 22:21:27.361001 I | raft: 18726fe191b3281c [logterm: 2, index: 8] sent MsgVote request to ff14b166cbcbf7e at term 3
2017-05-18 22:21:27.361096 I | raft: 18726fe191b3281c [logterm: 2, index: 8] sent MsgVote request to a59d378281242d56 at term 3
2017-05-18 22:21:27.568356 I | etcdserver/api/v3rpc: grpc: addrConn.transportMonitor exits due to: context canceled
2017-05-18 22:21:27.568549 I | etcdserver/api/v3rpc: grpc: addrConn.transportMonitor exits due to: context canceled
2017-05-18 22:21:27.568995 I | integration: terminating 5758625247135631596 (unix://localhost:57586252471356315960)
2017-05-18 22:21:27.569538 I | integration: terminated 5758625247135631596 (unix://localhost:57586252471356315960)
2017-05-18 22:21:27.569682 I | integration: terminating 1024999453306919239 (unix://localhost:10249994533069192390)
2017-05-18 22:21:27.569904 I | integration: terminating 4860400103667807978 (unix://localhost:48604001036678079780)
2017-05-18 22:21:27.570343 I | integration: terminated 4860400103667807978 (unix://localhost:48604001036678079780)
2017-05-18 22:21:27.571283 I | rafthttp: stopping peer a59d378281242d56...
2017-05-18 22:21:27.572754 I | rafthttp: closed the TCP streaming connection with peer a59d378281242d56 (stream MsgApp v2 writer)
2017-05-18 22:21:27.572797 I | rafthttp: stopped streaming with peer a59d378281242d56 (writer)
2017-05-18 22:21:27.573009 I | rafthttp: stopped streaming with peer a59d378281242d56 (writer)
2017-05-18 22:21:27.573139 I | rafthttp: stopped HTTP pipelining with peer a59d378281242d56
2017-05-18 22:21:27.573204 I | rafthttp: stopped streaming with peer a59d378281242d56 (stream MsgApp v2 reader)
2017-05-18 22:21:27.573275 I | rafthttp: stopped streaming with peer a59d378281242d56 (stream Message reader)
2017-05-18 22:21:27.573328 I | rafthttp: stopped peer a59d378281242d56
2017-05-18 22:21:27.573360 I | rafthttp: stopping peer ff14b166cbcbf7e...
2017-05-18 22:21:27.574728 I | rafthttp: closed the TCP streaming connection with peer ff14b166cbcbf7e (stream MsgApp v2 writer)
2017-05-18 22:21:27.574772 I | rafthttp: stopped streaming with peer ff14b166cbcbf7e (writer)
2017-05-18 22:21:27.574942 I | rafthttp: stopped streaming with peer ff14b166cbcbf7e (writer)
2017-05-18 22:21:27.575039 I | rafthttp: stopped HTTP pipelining with peer ff14b166cbcbf7e
2017-05-18 22:21:27.575102 I | rafthttp: stopped streaming with peer ff14b166cbcbf7e (stream MsgApp v2 reader)
2017-05-18 22:21:27.575175 I | rafthttp: stopped streaming with peer ff14b166cbcbf7e (stream Message reader)
2017-05-18 22:21:27.575218 I | rafthttp: stopped peer ff14b166cbcbf7e
2017-05-18 22:21:27.578308 I | integration: terminated 1024999453306919239 (unix://localhost:10249994533069192390)
--- FAIL: TestV3LeaseRequireLeader (1.28s)
        v3_lease_test.go:539: [DEBUG] FAIL; did not receive leader loss error first (took 823.437254ms)
```

Should we just wait more?

**Comment by heyitsanthony** [comment]
yeah we can bump the timeout
